core/table.ts

```diff
@@ -1,63 +1,165 @@
+  /**
+   * Sets the DISTKEY property when creating tables.
+   *
+   * For more information, read the AWS documentation [here](https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_examples.html#r_CREATE_TABLE_NEW-diststyle-distkey-and-sortkey-options).
+   */
+  distKey?: string;
+  distStyle?: string;
+  sortKeys?: string[];
+  sortStyle?: string;
+  bind?: boolean;
 }
-export enum DistStyleTypes {
-  EVEN = "even",
-  KEY = "key",
-  ALL = "all"
+
+/**
+ * Options for creating tables within Azure SQL Datawarehouse projects.
+ */
+export interface ISQLDataWarehouseOptions {
+  /**
+   * The distribution option value.
+   *
+   * For more information, read the Azure documentation [here](https://docs.microsoft.com/en-gb/sql/t-sql/statements/create-table-as-select-azure-sql-data-warehouse?view=aps-pdw-2016-au7#examples-for-table-distribution).
+   */
+  distribution?: string;
 }
-export enum SortStyleTypes {
-  COMPOUND = "compound",
-  INTERLEAVED = "interleaved"
+
+/**
+ * Options for creating tables within BigQuery projects.
+ */
+export interface IBigQueryOptions {
+  /**
+   * The key for partitioning the table by, typically a timestamp or date.
+   *
+   * For more information, read the BigQuery documentation [here](https://cloud.google.com/bigquery/docs/partitioned-tables#next_steps).
+   */
+  partitionBy?: string;
 }
 
-export const ignoredProps: {
-  [tableType: string]: Array<keyof dataform.ITable>;
-} = {
-  [TableTypes.INLINE]: [
-    "bigquery",
-    "redshift",
-    "sqlDataWarehouse",
-    "preOps",
-    "postOps",
-    "actionDescriptor",
-    "disabled",
-    "where"
-  ]
-};
-
-type ValueOf<T> = T[keyof T];
-export type TContextable<T> = T | ((ctx: TableContext) => T);
-export type TableType = ValueOf<TableTypes>;
-
-export interface TConfig {
+/**
+ * General options that can be provided to a table.
+ */
+export interface ITableConfig extends ICommonOutputConfig {
+  /**
+   * The type of the table. For more information, check out the documentation [guides](guides).
+   */
   type?: TableType;
-  dependencies?: Resolvable | Resolvable[];
-  tags?: string[];
-  description?: string;
-  columns?: IColumnsDescriptor;
+
+  /**
+   * When set to true, the action and SQL queries will not be executed, however they will still be compiled
+   * and the action can still be dependend upon. Useful for temporarily turning off broken actions.
+   */
   disabled?: boolean;
+
+  /**
+   * Only allowed when the table type is <code>"incremental"</code>.
+   * When set to true, the full-refresh option will have no effect. This is useful for tables that
+   * Are built from datasources that are transient, and makes sure historical data can never be lost.
+   */
   protected?: boolean;
-  redshift?: dataform.IRedshiftOptions;
-  bigquery?: dataform.IBigQueryOptions;
-  sqldatawarehouse?: dataform.ISQLDataWarehouseOptions;
-  database?: string;
-  schema?: string;
+
+  /**
+   * Redshift specific warehouse options.
```

Comment:
These should probably be hyphenated, `Redshift-specific`, (for the below options too)